aws/data_source_aws_dx_connection.go

```diff
@@ -0,0 +1,81 @@
+package aws
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/directconnect"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsDxConnection() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsDxConnectionRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"location": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"bandwidth": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"jumbo_frame_capable": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
Add `"tags": tagsSchemaComputed()` (and query for tags) as DX Connection now support resource tags.
Maybe set an `"arn"` attribute as well?